### ISSUE TITLE
Sharing Sketches

### ISSUE THREAD

**Issue by noio** [davidedc/livecodelab#212]
### Sharing

Allow users to share sketches by just dumping them to Gist or similar. Allow other users to **import** these sketches into their own sketch: `import gist/123456`. The imported sketch could consist of fully fleshed out visuals, or it could just define functions (#211) (like `grid`) with macros, so the user now has access to these extras.
#### Other thoughs:
- **Curation**: there could be a list of curated "standard" functions that are included in some dropdown from the LCL menu. This would also be a place to define aliases, e.g. `import grid --> import gist/123456`.
- `ifmain`: the imported sketches might need a way of knowing whether they are viewed as a standalone sketch or imported. If the sketch is viewed by itself, it could include examples of the function that was defined.

**Comment by Richard-Mineall** [comment]
Sharing sketches could just be a simple save and load feature, you could save your own sketches and choose to make them public which we could send to reddit or here to github, as all you need to do is copy and paste the code that you have written as a text file, people could download them as a file that can be opened directly by livecodelab, then open them at any time they want to using the load feature that I was talking about.

**Comment by davidedc** [comment]
I'd say let's keep it simple as first step: click share button - the code is saved in the URL and the URL is minified. That's it. Wanna share it? Share it. Wanna save it? Bookmark it. That's how Mr Doob does it here: http://mrdoob.com/#/146/html_editor

**Comment by Richard-Mineall** [comment]
yes, but where do the codes that are shared go? To a server most likely, but we want to do something else, so I'm saying that sharing it would post it to reddit and it would show up in a "shared" tab in livecodelab, what I'm saying basically is that we use an existing share website (with the ability to show symbols and indents) as a place to put the code without having to use a server.

**Comment by davidedc** [comment]
yeah yeah I'm cool with "fully fledged" server-heavy sharing - I made myself already a fully working system here https://github.com/davidedc/sketchPatch , which also saves thumbnails, has comments, manages permissions, "forks" sketches, tags, shows by author, etc.

I'm just saying what the first step is to make the simplest URL-encoded sharing.

Waiting on big releases doesn't make too much sense, I suggest we ship well-made and robust small stuff first, and then we iterate...

**Comment by Joshfindit** [comment]
How about just overloading the URL?

It would be cumbersome for larger sketches, but it could also mean that “run” links could be added to the screenshots on the site without having to have any server-side code.

Even the [safest limit is 2000](https://stackoverflow.com/questions/812925/what-is-the-maximum-possible-length-of-a-query-string) characters which is a decent amount of code.

As well, code that’s added in-editor could simply be echoed to the url meaning users could copy/paste exactly as they expect.
